src/System.Runtime.Extensions/src/System/BitConverter.cs

```diff
@@ -364,20 +364,39 @@ public static string ToString(byte[] value, int startIndex, int length)
             }
 
             int chArrayLength = length * 3;
+            const int StackLimit = 512; // arbitrary limit to switch from stack to heap allocation
```

Comment:
I could half-remember that there was some case in which it could be caught, but remember now that that case is when it was explicitly thrown, so not applicable.